docs/aurora/data-science/frameworks/dask.md

```diff
@@ -0,0 +1,229 @@
+
+
+import time
+import dpnp as np
+
+
+def count_points_inside_circle(N):
+    x = np.random.uniform(low=-1.0, high=1.0, size=(N, 2))
+    inside_circle = ((x * x).sum(axis=1) < 1.).sum()
+    return int(inside_circle)
+
+
+def compute_pi(inside_circle, N):
+    return 4 * inside_circle / N
+
+
+def run():
+    start = time.time()
+    num_workers = len(client.scheduler_info()['workers'])
+    N = 10_400_000_004
+
+    # number of points per worker
+    Neach_section, extras = divmod(N, num_workers)
+    points_per_worker = [Neach_section for _ in range(num_workers)]
+    points_per_worker[-1] += extras
+
+    futures = client.map(count_points_inside_circle, points_per_worker)
+    inside_circle = client.submit(sum, futures).result()
+    pi = compute_pi(inside_circle, N)
+    end = time.time()
+    return f"Num samples: {N:.2E}\t\tEstimate: {pi:.9f}\t\tTime taken: {end - start:.3f} s"
+
+
+def main(runs=5):
+    for i in range(runs):
+        print(f"Run {i}\t\t{run()}")
+
+
+main()
+client.shutdown()
+```
+
+### Run the `pi_dask_gpu.py` example
+
+- First, request an interactive job on 1 node.
+- Then, [start a Dask cluster with 6 GPU workers](#start-a-cluster-with-gpu-workers) and wait about 10 seconds for the cluster to start.
+- Press `Ctrl+Z` to send the process to the background, or open a new shell and ssh into the compute node. 
+- Run the example script:
+  ```bash
+  module load frameworks
+  conda activate dask
+  python pi_dask_gpu.py
+  ```
+- Output:
+  ```bash
+  <bound method Client.scheduler_info of <Client: 'tcp://10.168.0.10:8786' processes=6 threads=204, memory=1.00 TiB>>
+  Run 0           Num samples: 1.04E+10           Estimate: 3.141653798           Time taken: 1.596 s
+  Run 1           Num samples: 1.04E+10           Estimate: 3.141570887           Time taken: 1.354 s
+  Run 2           Num samples: 1.04E+10           Estimate: 3.141651954           Time taken: 1.451 s
+  Run 3           Num samples: 1.04E+10           Estimate: 3.141636617           Time taken: 0.518 s
+  Run 4           Num samples: 1.04E+10           Estimate: 3.141650108           Time taken: 0.511 s
+  ```
+
+## Connect to a Dask cluster from JupyterLab
+
+Here are the steps to start a Dask cluster and connect to it interactively from a Jupyter notebook:
+
+- First, request an interactive job on 1 node. Print the compute node's hostname (that you get with the command `hostname`), which will be used later.
+- Then, [start a Dask cluster](#start-a-cluster-with-gpu-workers) and wait about 10 seconds for the cluster to start.
+- **On your local machine**, open a ssh tunnel to the compute node (`COMPUTE_NODE` is the compute node's hostname and `YOUR_ALCF_USERNAME` is your ALCF username):
+  ```bash
+  ssh -t -L 23456:localhost:23456 -L 8787:localhost:8787 YOUR_ALCF_USERNAME@bastion.alcf.anl.gov ssh -t -L 23456:localhost:23456 -L 8787:localhost:8787 login.aurora.alcf.anl.gov ssh -t -L 23456:localhost:23456 -L 8787:localhost:8787 COMPUTE_NODE
```

Comment:
I get nervous when I see this multi-hop SSH tunnel line in our tutorials and docs, as there are several pitfalls as we saw during the `scikit-learn` hands-on exercise back in October: https://github.com/argonne-lcf/ALCF_Hands_on_HPC_Workshop/issues/56

Are any of those pitfalls possible here, i.e. should we suggest installing the SSH keypair in `~/.ssh/authorized_keys` for the compute node jump?